templates/helm/templates/secret.yaml

```diff
@@ -0,0 +1,19 @@
+{{- if .Values.deployment.secretEnvVars -}}
+apiVersion: v1
+kind: Secret
+metadata:
+  name: {{ include "app.fullname" . }}
+  namespace: {{ .Release.Namespace }}
+  labels:
+    app.kubernetes.io/name: {{ include "app.name" . }}
+    app.kubernetes.io/instance: {{ .Release.Name }}
+    app.kubernetes.io/managed-by: Helm
+    app.kubernetes.io/version: {{ .Chart.AppVersion | quote }}
+    k8s-app: {{ include "app.name" . }}
+    helm.sh/chart: {{ include "chart.name-version" . }}
+    control-plane: controller
+data:
+{{- range $key, $value := .Values.deployment.secretEnvVars }}
```

Comment:
I believe you want `secretEnvVars` to be of type `object` not `array`, otherwise $key above is going to be numeric indexes...

Reply:
🤦 you're right, will update